bokehjs/src/coffee/renderer/annotation/tooltip.coffee

```diff
@@ -54,17 +54,17 @@ define [
       @$el.removeClass('bk-right')
       @$el.removeClass('bk-left')
 
-      if side == "right"
-        @$el.addClass("bk-left")
-        top  = sy - @$el.height()/2
-        left = sx + 18
-      else if side == "left"
-        @$el.addClass("bk-right")
-        top  = sy - @$el.height()/2
-        left = sx - @$el.width() - 23
-      else
-        logger.warn("invalid tooltip side: '#{side}'")
-        return
+      arrow_width = 10
+
+      switch side
+        when "right"
+          @$el.addClass("bk-left")
+          left = sx + (@$el.outerWidth() - @$el.innerWidth()) + arrow_width
+        when "left"
+          @$el.addClass("bk-right")
+          left = sx - @$el.outerWidth() - arrow_width
+
+      top = sy - @$el.outerHeight()/2
```

Comment:
Because we shouldn't spoil our code with cases that shouldn't happen, just our language is so weak that it doesn't give us tools to express what we want. In a stronger programming language, `side` would be an enumeration at this point and we would just match both cases, and even get feedback if this mach is exhaustive or not.

On the other hand, I thought that validation of this was somewhere else as well, but now I don't see it, so perhaps this could be restored in some more appropriate location (e.g. `if` block at line 47).